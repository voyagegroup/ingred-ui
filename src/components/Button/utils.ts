import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
