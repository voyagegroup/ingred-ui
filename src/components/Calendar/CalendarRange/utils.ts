import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
