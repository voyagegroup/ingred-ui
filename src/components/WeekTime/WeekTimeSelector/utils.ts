const bin2hex = (bin: string) => {
  let out = "";
  for (let i = 0; i < bin.length; i += 4) {
    const c = bin.substr(i, 4);
    switch (c) {
      case "0000":
        out += "0";
        break;
      case "0001":
        out += "1";
        break;
      case "0010":
        out += "2";
        break;
      case "0011":
        out += "3";
        break;
      case "0100":
        out += "4";
        break;
      case "0101":
        out += "5";
        break;
      case "0110":
        out += "6";
        break;
      case "0111":
        out += "7";
        break;
      case "1000":
        out += "8";
        break;
      case "1001":
        out += "9";
        break;
      case "1010":
        out += "a";
        break;
      case "1011":
        out += "b";
        break;
      case "1100":
        out += "c";
        break;
      case "1101":
        out += "d";
        break;
      case "1110":
        out += "e";
        break;
      case "1111":
        out += "f";
        break;
      default:
        return "";
    }
  }

  return out;
};

export const convertTargetSettingToHex = (targetSetting: string[][]) => {
  let bin = "";
  for (let d = 0; d < 7; d++) {
    for (let h = 0; h < 24; h++) {
      bin += targetSetting[d][h];
    }
  }

  return bin2hex(bin);
};

/**
 * 開始位置と終了位置のそれぞれのの曜日と時間帯から、選択範囲を取得する
 */
export const getNewWeekTimeList = (
  startWeekIndex: number,
  startTimeIndex: number,
  endWeekIndex: number,
  endTimeIndex: number,
  newValue: string,
  weekTimeList: string[][],
) => {
  const newWeekTimeList = [...weekTimeList];
  const minWeekIndex = Math.min(startWeekIndex, endWeekIndex);
  const maxWeekIndex = Math.max(startWeekIndex, endWeekIndex);
  const minTimeIndex = Math.min(startTimeIndex, endTimeIndex);
  const maxTimeIndex = Math.max(startTimeIndex, endTimeIndex);

  for (let i = minWeekIndex; i <= maxWeekIndex; i++) {
    const weekTime = newWeekTimeList[i];
    const newWeekTime = weekTime.map((t, index) => {
      if (
        i >= minWeekIndex &&
        i <= maxWeekIndex &&
        index >= minTimeIndex &&
        index <= maxTimeIndex
      ) {
        return newValue; // 選択範囲内のセルの場合、newValueに更新
      }
      return t;
    });

    newWeekTimeList[i] = newWeekTime;
  }

  return newWeekTimeList;
};
