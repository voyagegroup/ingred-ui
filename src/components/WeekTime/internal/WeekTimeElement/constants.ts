// 0 to 24 array
export const timeList = Array.from(Array(24)).map((_, i) => i);
