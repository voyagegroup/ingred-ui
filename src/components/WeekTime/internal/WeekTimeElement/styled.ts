import styled from "styled-components";

export const WeekTimeContainer = styled.div`
  display: grid;
  grid-template-columns: 38px repeat(24, 1fr);
  grid-gap: ${({ theme }) => theme.spacing / 2}px;
  align-items: center;
  justify-items: center;
  width: fit-content;
`;

export const ContainerBase = styled.div`
  height: 24px;
  text-align: center;
  background-color: ${({ theme }) => theme.palette.gray.light};
  border-radius: ${({ theme }) => theme.spacing / 2}px;
  border: 1px solid ${({ theme }) => theme.palette.gray.main};
`;

export const WeekContainer = styled(ContainerBase)`
  width: 38px;
`;

export const TimeContainer = styled(ContainerBase)`
  width: 24px;
`;

export const EmptyContainer = styled(ContainerBase)`
  border: none;
`;
