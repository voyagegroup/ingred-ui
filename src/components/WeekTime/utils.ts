export const hex2bin = (hex: string) => {
  if (!hex) return Array(7).fill(Array(24).fill("1")).join("");

  let out = "";
  for (const c of hex) {
    switch (c) {
      case "0":
        out += "0000";
        break;
      case "1":
        out += "0001";
        break;
      case "2":
        out += "0010";
        break;
      case "3":
        out += "0011";
        break;
      case "4":
        out += "0100";
        break;
      case "5":
        out += "0101";
        break;
      case "6":
        out += "0110";
        break;
      case "7":
        out += "0111";
        break;
      case "8":
        out += "1000";
        break;
      case "9":
        out += "1001";
        break;
      case "a":
        out += "1010";
        break;
      case "b":
        out += "1011";
        break;
      case "c":
        out += "1100";
        break;
      case "d":
        out += "1101";
        break;
      case "e":
        out += "1110";
        break;
      case "f":
        out += "1111";
        break;
      default:
        return "";
    }
  }

  return out;
};

/**
 * @param hex odr_targeting_week_time の16進数表記
 * @returns 週ごとの時間帯ターゲティングの設定の多重配列。[曜日][時間]の順（0番目が月曜日、0番目のn番目がn時）
 */
export const getTargetSetting = (hex?: string): string[][] => {
  if (!hex) return Array(7).fill(Array(24).fill("1"));

  const binary = hex2bin(hex);
  const times: string[][] = [];

  for (let d = 0; d < 7; d++) {
    times[d] = [];
    for (let h = 0; h < 24; h++) {
      times[d][h] = binary.charAt(d * 24 + h);
    }
  }

  return times;
};
