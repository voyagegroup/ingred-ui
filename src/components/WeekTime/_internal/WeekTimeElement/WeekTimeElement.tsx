import * as Styled from "./styled";
import { useLocaleProps } from "../../../../hooks/useLocaleProps";
import React, { Fragment, useMemo } from "react";
import { getTargetSetting } from "../../utils";
import { timeList } from "../../constants";

export type WeekTimeElementProps = {
  weekTime: string;
  isWithinHoverRange?: (weekIndex: number, timeIndex: number) => boolean;
  onMouseOver?: (weekIndex: number, timeIndex: number) => void;
  onMouseDown?: (weekIndex: number, timeIndex: number, time: string) => void;
};

export const WeekTimeElement: React.FC<WeekTimeElementProps> = (inProps) => {
  const props = useLocaleProps({
    props: inProps,
    name: "WeekTimeElement",
  });
  const { weekList, weekTime, isWithinHoverRange, onMouseOver, onMouseDown } =
    props;
  const weekTimeList = useMemo(() => getTargetSetting(weekTime), [weekTime]);
  const defaultHandler = () => {};

  return (
    <Styled.WeekTimeContainer>
      <Styled.EmptyContainer />
      {timeList.map((time) => (
        <Styled.TimeContainer key={time}>{time}</Styled.TimeContainer>
      ))}
      {weekTimeList.map((time, weekIndex) => (
        // eslint-disable-next-line react/no-array-index-key
        <Fragment key={weekIndex}>
          <Styled.WeekContainer>{weekList[weekIndex]}</Styled.WeekContainer>
          {time.map((t, timeIndex) => (
            <Styled.WeekTimeItem
              // eslint-disable-next-line react/no-array-index-key
              key={`${weekIndex}-${timeIndex}`}
              active={t === "1"}
              as={isWithinHoverRange ? "button" : "span"}
              hover={
                isWithinHoverRange
                  ? isWithinHoverRange(weekIndex, timeIndex)
                  : false
              }
              onMouseOver={
                onMouseOver?.(weekIndex, timeIndex) ?? defaultHandler
              }
              onMouseDown={
                onMouseDown?.(weekIndex, timeIndex, t) ?? defaultHandler
              }
            />
          ))}
        </Fragment>
      ))}
    </Styled.WeekTimeContainer>
  );
};
