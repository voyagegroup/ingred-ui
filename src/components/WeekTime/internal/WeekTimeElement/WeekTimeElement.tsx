import * as Styled from "./styled";
import React, { Fragment, useMemo } from "react";
import { getTargetSetting } from "../../utils";
import { timeList } from "./constants";

export type WeekTimeElementProps = {
  weekTime: string;
  weekList: string[];
  WeekTimeItem: React.FC<
    { active: boolean; hover: boolean } & React.HTMLAttributes<
      HTMLButtonElement | HTMLSpanElement
    >
  >;
  isWithinHoverRange?: (weekIndex: number, timeIndex: number) => boolean;
  onMouseOver?: (weekIndex: number, timeIndex: number) => void;
  onMouseDown?: (weekIndex: number, timeIndex: number, time: string) => void;
};

export const WeekTimeElement: React.FC<WeekTimeElementProps> = ({
  weekList,
  weekTime,
  WeekTimeItem,
  isWithinHoverRange,
  onMouseOver,
  onMouseDown,
}) => {
  const weekTimeList = useMemo(() => getTargetSetting(weekTime), [weekTime]);

  const handleMouseOver = (weekIndex: number, timeIndex: number) => () => {
    onMouseOver?.(weekIndex, timeIndex);
  };

  const handleMouseDown =
    (weekIndex: number, timeIndex: number, time: string) => () => {
      onMouseDown?.(weekIndex, timeIndex, time);
    };

  return (
    <Styled.WeekTimeContainer>
      <Styled.EmptyContainer />
      {timeList.map((time) => (
        <Styled.TimeContainer key={time}>{time}</Styled.TimeContainer>
      ))}
      {weekTimeList.map((time, weekIndex) => (
        // eslint-disable-next-line react/no-array-index-key
        <Fragment key={weekIndex}>
          <Styled.WeekContainer>{weekList[weekIndex]}</Styled.WeekContainer>
          {time.map((t, timeIndex) => (
            <WeekTimeItem
              // eslint-disable-next-line react/no-array-index-key
              key={`${weekIndex}-${timeIndex}`}
              active={t === "1"}
              hover={
                isWithinHoverRange
                  ? isWithinHoverRange(weekIndex, timeIndex)
                  : false
              }
              onMouseOver={handleMouseOver(weekIndex, timeIndex)}
              onMouseDown={handleMouseDown(weekIndex, timeIndex, t)}
            />
          ))}
        </Fragment>
      ))}
    </Styled.WeekTimeContainer>
  );
};
