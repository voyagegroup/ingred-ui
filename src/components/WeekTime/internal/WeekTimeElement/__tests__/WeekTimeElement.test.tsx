import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../../../utils/renderWithThemeProvider";
import { WeekTimeElement } from "../WeekTimeElement";

describe("WeekTime component testing", () => {
  afterEach(cleanup);

  test("WeekTime", () => {
    const { asFragment } = renderWithThemeProvider(
      <WeekTimeElement weekTime="ffffffffffffffffffffffffffffffffffffffffff" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
