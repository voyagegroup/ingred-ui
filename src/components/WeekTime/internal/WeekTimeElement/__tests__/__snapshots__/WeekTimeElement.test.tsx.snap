// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekTime component testing WeekTime 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ bvzRRH"
  >
    <div
      class="sc-gsnTZi sc-eCYdqJ guZhMt cKEPO"
    />
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      0
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      1
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      2
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      3
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      4
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      5
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      6
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      7
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      8
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      9
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      10
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      11
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      12
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      13
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      14
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      15
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      16
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      17
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      18
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      19
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      20
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      21
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      22
    </div>
    <div
      class="sc-gsnTZi sc-hKMtZM guZhMt eoKgCV"
    >
      23
    </div>
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Sun
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Mon
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Tue
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Wed
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Thu
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Fri
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <div
      class="sc-gsnTZi sc-dkzDqf guZhMt gWrkHL"
    >
      Sat
    </div>
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
    <span
      class="sc-jSMfEi dnQWRT"
    />
  </div>
</DocumentFragment>
`;
